File: clients/CSharp/DebugInterface.cs
using System.IO;

namespace Aicup2020
{
    public class DebugInterface
    {
        private BinaryWriter writer;
        private BinaryReader reader;
        public DebugInterface(BinaryReader reader, BinaryWriter writer)
        {
            this.reader = reader;
            this.writer = writer;
        }
        public void Send(Model.DebugCommand command)
        {
            new Model.ClientMessage.DebugMessage(command).WriteTo(writer);
            writer.Flush();
        }
        public Model.DebugState GetState()
        {
            new Model.ClientMessage.RequestDebugState().WriteTo(writer);
            writer.Flush();
            return Model.DebugState.ReadFrom(reader);
        }
    }
}

File: clients/FSharp/DebugInterface.fs
namespace Aicup2020

type DebugInterface(reader, writer) =
    member this.send(command) =
        (Model.ClientMessage.DebugMessage { Command = command }).writeTo writer
        writer.Flush()

    member this.getState(): Model.DebugState =
        (new Model.ClientMessageRequestDebugState()).writeTo writer
        writer.Flush()
        Model.DebugState.readFrom reader
